pkg/steps/git_source.go

```diff
@@ -0,0 +1,65 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+
+	buildapi "github.com/openshift/api/build/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+
+	"github.com/openshift/ci-operator/pkg/api"
+)
+
+type gitSourceStep struct {
+	config      api.ProjectDirectoryImageBuildStepConfiguration
+	resources   api.ResourceConfiguration
+	imageClient imageclientset.ImageV1Interface
+	buildClient BuildClient
+	jobSpec     *api.JobSpec
+}
+
+func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return s.jobSpec.Inputs(), nil
+}
+
+func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
+	return handleBuild(s.buildClient, buildFromSource(s.jobSpec, "", s.config.To, buildapi.BuildSource{
+		Type:       buildapi.BuildSourceGit,
+		ContextDir: s.config.ContextDir,
+		Git: &buildapi.GitBuildSource{
+			URI: fmt.Sprintf("https://github.com/%s/%s.git", s.jobSpec.Refs.Org, s.jobSpec.Refs.Repo),
+			Ref: s.jobSpec.Refs.BaseRef,
+		},
+	}, s.config.DockerfilePath, s.resources), dry)
+}
+
+func (s *gitSourceStep) Done() (bool, error) {
+	return imageStreamTagExists(s.config.To, s.imageClient.ImageStreamTags(s.jobSpec.Namespace))
+}
+
+func (s *gitSourceStep) Name() string { return string(s.config.To) }
+
+func (s *gitSourceStep) Description() string {
+	return fmt.Sprintf("Build git source code into an image and tag it as %s", s.config.To)
+}
+
+func (s *gitSourceStep) Requires() []api.StepLink { return nil }
+
+func (s *gitSourceStep) Creates() []api.StepLink {
+	return []api.StepLink{api.InternalImageLink(s.config.To)}
+}
+
+func (s *gitSourceStep) Provides() (api.ParameterMap, api.StepLink) {
+	return nil, nil
```

Comment:
You need to fill these in

Reply:
Do we need this for the base image? We are not providing anything already.